### PR TITLE
return empty vault orders if clob pair does not exist or is in final settlement

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
when clob pair doesn't exist or is in final settlement, return empty orders and `nil` error

### Test Plan
unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for Clob pairs, allowing for smoother processing when non-existent pairs are encountered.
	- Introduced additional sanity checks to validate order data integrity.

- **Bug Fixes**
	- Adjusted test scenarios to reflect successful handling of previously error-prone cases, leading to more robust functionality.

- **Chores**
	- Updated error registration logic for improved clarity and consistency in error codes.

### CODE REVIEW

File: protocol/x/vault/types/errors.go [hunk 1]
```diff
@@ -10,84 +10,84 @@ var (
 		1,
 		"Shares are negative",
 	)
-	ErrClobPairNotFound = errorsmod.Register(
-		ModuleName,
-		2,
-		"ClobPair not found",
-	)
 	ErrMarketParamNotFound = errorsmod.Register(
```
Comment (vincentwschau) [inline protocol/x/vault/types/errors.go:13]: nit: Don't change error code numbering. Just remove deprecated errors / add new errors with increased error codes.

File: protocol/x/vault/keeper/orders_test.go [hunk 2]
```diff
@@ -995,13 +1015,12 @@ func TestGetVaultClobOrderIds(t *testing.T) {
...
 			layers:  0,
 		},
-		"Vault Clob 797 (non-existent clob pair), 2 layers": {
+		"Vault Clob 797, 2 layers": {
 			vaultId: vaulttypes.VaultId{
 				Type:   vaulttypes.VaultType_VAULT_TYPE_CLOB,
 				Number: 797,
 			},
-			layers:      2,
-			expectedErr: vaulttypes.ErrClobPairNotFound,
```
Comment (vincentwschau) [inline protocol/x/vault/keeper/orders_test.go:1004]: nit: Update test to test expected number of orders? This test case isn't clear what's being tested.

**Review by vincentwschau — APPROVED** [review]

**Comment by tqin7** [comment]
backport release/protocol/v6.x

**Comment by tqin7** [comment]
@Mergifyio backport release/protocol/v6.x

**Pull request merged**